lib/LaTeXML/Common/Color.pm

```diff
@@ -118,17 +118,34 @@ sub complement {
 
 # Mix $self*$fraction + $color*(1-$fraction)
 sub mix {
-  my ($self, $color, $fraction) = @_;
-  $color = $color->convert($self->model) unless $self->model eq $color->model;
-  my @a = $self->components;
-  my @b = $color->components;
-  return $self->new(map { $fraction * $a[$_] + (1 - $fraction) * $b[$_] } 0 .. $#a); }
+  my ($self, $other, $fraction) = @_;
+  my $base       = $self;
+  my $base_model = $base->model;
+  my $mix_model  = $other->model;
+  if ($base_model ne $mix_model) {
+    # mixing 'rgb|cmy' with 'gray' should yield 'rgb|cmy'.
+    if ($base_model eq 'gray') {
+      $base = $base->convert($mix_model);
+    } else {
+      $other = $other->convert($base_model);
+  } }
+  my @a = $base->components;
+  my @b = $other->components;
+  return $base->new(map { $fraction * $a[$_] + (1 - $fraction) * $b[$_] } 0 .. $#a); }
 
 sub add {
-  my ($self, $color) = @_;
-  $color = $color->convert($self->model) unless $self->model eq $color->model;
+  my ($self, $other) = @_;
+  my $base       = $self;
+  my $base_model = $base->model;
+  my $mix_model  = $other->model;
+  if ($base_model ne $mix_model) {
+    # mixing 'rgb|cmy' with 'gray' should yield 'rgb|cmy'.
+    if ($base_model eq 'gray') {
+      $base = $base->convert($mix_model);
+    } else {
+      $other = $other->convert($base_model); } }
   my @a = $self->components;
-  my @b = $color->components;
+  my @b = $other->components;
   return $self->new(map { $a[$_] + $b[$_] } 0 .. $#a); }
```

Comment:
shouldn't the last 2 `$self` be `$base`? (like for `mix`)

Reply:
Duh. How did that slip happen? One quote I overheard this morning:

"If it isn't tested, it is broken."

Reply:
Thank you for catching that and avoiding a very annoying encounter+report+debug cascade later on.